File: DataqSDKVB/DATAQSDK/Form1.vb
Public Class Form1
    Dim v(,) As Short
    Dim v2(,) As Short

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Start.Click
        AxDataqSdk1.DeviceDriver = TextBox1.Text
        AxDataqSdk1.DeviceID = TextBox2.Text
        AxDataqSdk1.SampleRate = Val(TextBox3.Text)
        AxDataqSdk1.ADChannelCount = 1
        AxDataqSdk1.EventPoint = 1

        AxDataqSdk1.Start()
        AxUltimaMeter2.MaxValue = AxDataqSdk1.SampleRate / 2
        TextBox3.Text = AxDataqSdk1.SampleRate
        Timer1.Interval = 10
        Timer1.Enabled = True
        Timer2.Interval = 50
        Timer2.Enabled = True
    End Sub

    Private Sub Button2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button2.Click
        AxDataqSdk1.Stop()
        Timer1.Enabled = False
        Timer2.Enabled = False
    End Sub

    Private Sub Timer1_Tick(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Timer1.Tick
        Dim dTemp As Double

        If AxDataqSdk1.AvailableData > 0 Then
            v = AxDataqSdk1.GetData
            AxXChart1.Chart(v)
            dTemp = Val(v(0, 0))
            AxUltimaMeter1.Value = dTemp / 3276.8

        End If

    End Sub
   

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        AxUltimaMeter1.MinValue = -10
        AxUltimaMeter1.MaxValue = 10
        AxUltimaMeter2.MinValue = 0
        AxUltimaMeter2.Unit = "Peak Hz"
        AxUltimaWaterfall1.ChartType = ULTIMAWATERFALLLib.enumCharType.uw3D
        AxUltimaWaterfall1.ChartDirection = ULTIMAWATERFALLLib.enumCharDir.uwNSlope
        AxUltimaWaterfall1.ScalingMethod = ULTIMAWATERFALLLib.enumColorScale.uwBGR

    End Sub



    Private Sub Timer2_Tick(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Timer2.Tick
        Dim i As Short
        v2 = AxDataqSdk1.GetDataFrame(1024)
        For i = 0 To 1023
            AxFFT10241.SetWaveformPt(i, v2(0, i))
        Next i
        AxFFT10241.FFTNow(1024)
        AxUltimaWaterfall1.Chart(AxFFT10241.Power)
        AxUltimaMeter2.Value = AxUltimaMeter2.MaxValue * (AxFFT10241.GetPeakIdx() / 512)
    End Sub

    
    Private Sub RadioButton1_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles RadioButton1.CheckedChanged
        If Timer1.Enabled Then AxUltimaWaterfall1.ChartDirection = ULTIMAWATERFALLLib.enumCharDir.uwB2T
    End Sub

    Private Sub CheckBox1_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles CheckBox1.CheckedChanged
        If Timer1.Enabled Then
            If CheckBox1.CheckState = CheckState.Checked Then
                AxUltimaWaterfall1.ChartType = ULTIMAWATERFALLLib.enumCharType.uw3D
                RadioButton6.Enabled = False
                RadioButton4.Enabled = False
            Else
                AxUltimaWaterfall1.ChartType = ULTIMAWATERFALLLib.enumCharType.uw2D
                RadioButton6.Enabled = True
                RadioButton4.Enabled = True
            End If
        End If
    End Sub

    Private Sub RadioButton6_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles RadioButton6.CheckedChanged
        If Timer1.Enabled Then AxUltimaWaterfall1.ChartDirection = ULTIMAWATERFALLLib.enumCharDir.uwR2Lx
    End Sub

    Private Sub RadioButton2_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles RadioButton2.CheckedChanged
        If Timer1.Enabled Then AxUltimaWaterfall1.ChartDirection = ULTIMAWATERFALLLib.enumCharDir.uwNSlope
    End Sub

    Private Sub RadioButton5_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles RadioButton5.CheckedChanged
        If Timer1.Enabled Then AxUltimaWaterfall1.ChartDirection = ULTIMAWATERFALLLib.enumCharDir.uwT2B
    End Sub

    Private Sub RadioButton3_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles RadioButton3.CheckedChanged
        If Timer1.Enabled Then AxUltimaWaterfall1.ChartDirection = ULTIMAWATERFALLLib.enumCharDir.uwSlope
    End Sub

    Private Sub RadioButton4_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles RadioButton4.CheckedChanged
        If Timer1.Enabled Then AxUltimaWaterfall1.ChartDirection = ULTIMAWATERFALLLib.enumCharDir.uwR2L
    End Sub
End Class

File: DataqSDKCS/DataqSDKCS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace DataqSDKCS
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            axDataqSdk1.DeviceDriver=textBox1.Text;

            axDataqSdk1.DeviceID=textBox2.Text;

            axUltimaMeter1.MaxValue=10;

            axUltimaMeter1.MinValue=-10;

 

            axDataqSdk1.SampleRate=Convert.ToDouble(textBox3.Text);

            axUltimaWaterfall1.ChartType = ULTIMAWATERFALLLib.enumCharType.uw3D;

            axUltimaWaterfall1.ChartDirection = ULTIMAWATERFALLLib.enumCharDir.uwSlope;

            axDataqSdk1.EventPoint=1;

            timer1.Interval=10;

            timer1.Enabled = true;

            timer2.Interval=50;

            timer2.Enabled = true; 

            axDataqSdk1.Start();
        }

        private void button2_Click(object sender, EventArgs e)
        {
            timer1.Enabled = false;
            timer2.Enabled = false;
            axDataqSdk1.Stop();
        }

        private void axDataqSdk1_NewData(object sender, AxDATAQSDKLib._DDataqSdkEvents_NewDataEvent e)
        {

        }

        private void timer1_Tick(object sender, EventArgs e)
        {
            if (axDataqSdk1.AvailableData > 0)
            {
                object V;
                short[,] intArray;
                V = axDataqSdk1.GetData();
                axXChart1.Chart(V);

                intArray = (short[,])V;
                axUltimaMeter1.Value = intArray[0, 0]/3276.8;
            }

        }

        private void timer2_Tick(object sender, EventArgs e)
        {
            object V;
            short[,] intArray;
            short i;

            V = axDataqSdk1.GetDataFrame(1024);
            intArray = (short[,])V;

            for (i=0; i<1024; i++){
                axFFT10241.SetWaveformPt(i, intArray[0, i]);
            }

            axFFT10241.FFTNow(1024);

            axUltimaWaterfall1.Chart(axFFT10241.Power());

        }

        private void axDataqSdk1_ControlError(object sender, AxDATAQSDKLib._DDataqSdkEvents_ControlErrorEvent e)
        {
            timer1.Enabled = false;
            timer2.Enabled = false;
            axDataqSdk1.Stop();
            MessageBox.Show("Error: Please check DeviceDriver and DeviceID");
        }
    }
}
